### PR TITLE
OCPBUGS-42422: Fix order rendering HCP objects

### PR DESCRIPTION
**Pull request by jparrill**
Fix order rendering HCP objects, modified to be always first the Namespace, and the last ones the HC and NP. Covered with unit tests.

**Which issue(s) this PR fixes**:
Fixes #[OCPBUGS-42422](https://issues.redhat.com/browse/OCPBUGS-42422)

### CODE REVIEW

File: cmd/cluster/core/create_test.go [hunk 1]
```diff
@@ -86,3 +89,61 @@ func TestValidateMgmtClusterAndNodePoolCPUArchitectures(t *testing.T) {
...
+			g := NewWithT(t)
+			objects := tc.resources.asObjects()
+			if tc.expectedFail {
+				g.Expect(objects[0]).To(Not(Equal(tc.resources.Namespace)))
+				return
+			}
+			g.Expect(objects[0]).To(Equal(tc.resources.Namespace), "Namespace should be the first object in the slice")
+			hcPosition := len(objects) - len(tc.resources.NodePools) - 1
+			g.Expect(objects[hcPosition]).To(Equal(tc.resources.Cluster), "HostedCluster should be the secodn-to-last object in the slice")
+			g.Expect(objects[len(objects)-1]).To(Equal(tc.resources.NodePools[1]), "NodePools should be the last object in the slice")
```
Comment (csrwng) [inline cmd/cluster/core/create_test.go:146]: s/`tc.resources.NodePools[1]`/`tc.resources.NodePools[len(tc.resources.NodePools)-1]`/

**Comment by jparrill** [comment]
Rebase with new konflux images

**Pull request merged**